en/test_and_ci/docker.md

```diff
@@ -129,10 +132,18 @@ docker run -it --privileged \
 -v ~/src/PX4-Autopilot:/src/PX4-Autopilot/:rw \
 -v /tmp/.X11-unix:/tmp/.X11-unix:ro \
 -e DISPLAY=:0 \
--p 14570:14570/udp \
---name=mycontainer px4io/px4-dev-ros:2017-10-23 bash
+--network host \
+--name=px4-ros px4io/px4-dev-ros2-foxy:2022-07-31 bash
 ```
 
+:::note
+We use the host network mode to avoid conflicts between the UDP port access control when using QGroundControl on the same system as the docker container.
+:::
+
+:::note
+In some situations, `DISPLAY` may need to be set to a different value such as `:1`. It may be possible to determine what that value is by using `echo $DISPLAY` or simply by changing `-e DISPLAY=:0` to `-e DISPLAY=$DISPLAY`. Note that making that change assumes that the environment variable `DISPLAY` has been set.
```

Comment:
Hi @magicbycalvin,

What the `-e` flag does (I think) is set environment flags in the container. Since we're passing in a value for DISPLAY from the host, I assume that what we're doing here is telling the container what display to use in the host.

Now the  $DISPLAY` environment variable is a magic number set by the XWindow system (according to the internet). So I would expect this to work any XWindow system like Ubuntu hosts, but not WIndows.

Further, what are the symptoms of getting the wrong display setting? Is it that gazebo might not display in the host machine, or something else?

Depending on the symptom I think we should reword like this:

```suggestion
If [symptoms of wrong display], `DISPLAY` may need to be set to a different value.
On Linux (XWindow) hosts you can change `-e DISPLAY=:0` to `-e DISPLAY=$DISPLAY`.
On other hosts you might iterate the value of `0` in  `-e DISPLAY=:0` until [symptom is fixed]
```

Reply:
Good point about Windows environments (I didn't even know you could get PX4 working on Windows!)
I agree with your suggested change and went ahead and copied that over. The symptom of a wrong display is that you get the error `Can't open display: :0`. I included that message in the [symptom of wrong display] and [symptom is fixed] comments above.